src/include/processor/operator/persistent/reader/csv/csv_error.h

```diff
@@ -0,0 +1,91 @@
+#pragma once
+
+#include <set>
+#include <string>
+#include <vector>
+
+#include "common/uniq_lock.h"
+
+namespace kuzu::processor {
+
+class BaseCSVReader;
+
+struct LineContext {
+    uint64_t startByteOffset;
+    uint64_t endByteOffset;
+
+    bool isCompleteLine;
+
+    void setNewLine(uint64_t start);
+    void setEndOfLine(uint64_t end);
+};
+
+// If parsing in parallel during parsing we may not be able to determine line numbers
+// Thus we have additional fields that can be used to determine line numbers + reconstruct lines
+// After parsing this will be used to populate a PopulatedCSVError instance
+struct CSVError {
+    std::string message;
+    std::string filePath;
+    LineContext errorLine;
+
+    uint64_t blockIdx;
+    uint64_t numRowsReadInBlock;
+
+    bool operator<(const CSVError& o) const;
+};
+
+struct PopulatedCSVError {
+    std::string message;
+    std::string filePath;
+    std::string reconstructedLine;
+    uint64_t lineNumber;
+};
+
+struct WarningCounter {
+    uint64_t count;
```

Comment:
If it is simply a struct of int, let's replace it with `using warning_counter = uint64;`

Reply:
updated